concent_api/concent_api/middleware.py

```diff
@@ -39,3 +44,34 @@ def __call__(self, request):
         response = self.get_response(request)
         response['Concent-Version'] = self._concent_version
         return response
+
+
+class HandleServerErrorMiddleware(object):
+    """
+    Used to catch all unhandled exceptions and return JSON response containing error information.
+    """
+    def __init__(self, get_response):
+        self.get_response = get_response
+        self.sid = None
+
+    def __call__(self, request):
+        self.sid = transaction.savepoint()
+        response = self.get_response(request)
+        transaction.savepoint_commit(self.sid)
+        return response
+
+    def process_exception(self, request, exception):  # pylint: disable=unused-argument
+        transaction.savepoint_rollback(self.sid)
+        return self._build_json_response(exception)
+
+    @staticmethod
+    def _build_json_response(exception):
+
+        message = {
+            "error_message": str(exception) or DEFAULT_ERROR_MESSAGE,
+            "error_code": getattr(exception, "error_code", "crash"),
```

Comment:
`crash` error_code should be defined along with other error codes and imported from there. Btw. I think crash is not best name here, how about something like `unhandled_error` ?

Reply:
ok, I will introduce additional `error_code`. "crash" is taken from @cameel 's comment https://github.com/golemfactory/concent/pull/328#discussion_r184674933

Reply:
Ok, I see you have added the code but you need to use it here too. I mean, `from utils.constants import ErrorCode` and replace `"crash"` with `ErrorCode.CONCENT_APPLICATION_CRASH.value`.

Reply:
sorry, missed that.

done!